lumeer-storage/lumeer-storage-api/src/main/java/io/lumeer/storage/api/dao/UserDao.java

```diff
@@ -18,28 +18,23 @@
  */
 package io.lumeer.storage.api.dao;
 
-import io.lumeer.api.model.Organization;
 import io.lumeer.api.model.User;
 
 import java.util.List;
 import java.util.Optional;
 
 public interface UserDao {
 
-   void createUsersRepository(Organization organization);
+   User createUser(String organizationId, User user);
 
-   void deleteUsersRepository(Organization organization);
+   User updateUser(String organizationId, String userId, User user);
 
-   void setOrganization(Organization organization);
+   void deleteUser(String organizationId, String userId);
```

Comment:
Why do we need `organizationId` parameter on these three methods? It is not needed here since the DAO layer should only be responsible for storing and retrieving data. And it is system-scoped so there is no need for an organization ID here.

Reply:
Look at how users are stored in DB. How can I know, which embedded document from array of groups should I delete?

Reply:
Yes, but you don't need it on this level. I mentioned it in the other comment as well. This business logic needs to be in the facade not in DAO.

Deleting user on DAO level means deleting it from the database. (Which is not really something we need at the moment.)

Reply:
Maybe I should rename method name to `deleteUserFromOrganization`, but without this parameter `dao` layer doesn't know how to build correct request do db.

Reply:
And what about looking at this from another perspective? DAO is only responsible for simple data operations. If you want to have a method like `deleteUserFromOrganization`, it should be on the facade level and calling `updateUser` or something similar.

Or it can be like you suggest but make sure there absolutely no business logic involved in the DAO layer. All of it needs to be in the facade and DAO methods should be called there only where you would put a single database operation if you used MongoDB driver directly in the facades.